src/InMemoryEventStore.php

```diff
@@ -0,0 +1,306 @@
+                $this->streams[$streamName->toString()]['metadata']
+            ));
+        });
+
+        $actionEventEmitter->attachListener(self::EVENT_LOAD_REVERSE, function (ActionEvent $event): void {
+            $streamName = $event->getParam('streamName');
+            $fromNumber = $event->getParam('fromNumber');
+            $count = $event->getParam('count');
+            $metadataMatcher = $event->getParam('metadataMatcher');
+
+            if (! $this->hasStream($streamName)) {
+                $event->stopPropagation();
+                return;
+            }
+
+            if (null === $metadataMatcher) {
+                $metadataMatcher = new MetadataMatcher();
+            }
+
+            $streamEvents = [];
+
+            foreach ($this->streams[$streamName->toString()]['events'] as $key => $streamEvent) {
+                if ($this->matchesMetadata($metadataMatcher, $streamEvent->metadata())
+                    && ((null === $count
+                            && ($key + 1) <= $fromNumber
+                        ) || (null !== $count
+                            && ($key + 1) <= $fromNumber
+                            && ($key + 1) >= ($fromNumber - $count + 1)
+                        )
+                    )
+                ) {
+                    $streamEvents[] = $streamEvent;
+                }
+            }
+
+            $streamEvents = array_reverse($streamEvents);
+
+            $event->setParam('stream', new Stream(
+                $streamName,
+                new ArrayIterator($streamEvents),
+                $this->streams[$streamName->toString()]['metadata']
+            ));
+        });
+    }
+
+    public function getActionEventEmitter(): ActionEventEmitter
+    {
+        return $this->actionEventEmitter;
+    }
+
+    public function hasStream(StreamName $streamName): bool
+    {
+        return isset($this->streams[$streamName->toString()]);
+    }
+
+    public function fetchStreamMetadata(StreamName $streamName): array
+    {
+        if (! $this->hasStream($streamName)) {
+            throw StreamNotFound::with($streamName);
+        }
+
+        return $this->streams[$streamName->toString()]['metadata'];
+    }
+
+    public function create(Stream $stream): void
+    {
+        $argv = ['stream' => $stream, 'streamEvents' => $stream->streamEvents()];
+
+        $event = $this->actionEventEmitter->getNewActionEvent(self::EVENT_CREATE, $this, $argv);
+
+        $this->actionEventEmitter->dispatch($event);
+
+        if (! $event->getParam('result', false)) {
+            throw StreamExistsAlready::with($stream->streamName());
+        }
+    }
+
+    public function appendTo(StreamName $streamName, Iterator $streamEvents): void
+    {
+        $argv = ['streamName' => $streamName, 'streamEvents' => $streamEvents];
+
+        $event = $this->actionEventEmitter->getNewActionEvent(self::EVENT_APPEND_TO, $this, $argv);
+
+        $this->getActionEventEmitter()->dispatch($event);
+
```

Comment:
@prolic Any reasons why `$this->actionEventEmitter` and `$this->getActionEventEmitter()` is used here? We should use `$this->actionEventEmitter`.